### ISSUE TITLE
TrustedHostMiddleware ignores port in Host header

### ISSUE THREAD

**Issue by blipk** [tiangolo/fastapi#5818]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()
app.add_middleware(
    TrustedHostMiddleware,
    allowed_hosts=["localhost:8000"]
)
```

### Description

Using the provided example code and making a request from localhost:8000 I get an "Invalid host header" response, even though looking at dev tools in chrome the header is `Host: localhost:8000`

If I change to `allowed_hosts=["localhost"]` it works fine.

I understand it's a trusted host, and not a trusted port, but seems strange to me that it says invalid host header even though thats exactly what the browser is sending in the host header.

Not sure if this is intended behaviour or not.

### Operating System

Linux

### FastAPI Version

0.88.0

### Python Version

3.10.8

**Comment by iudeen** [comment]
I don't think you can have port as a part of host. Host names do not contain port🤔

**Comment by aybruhm** [comment]
> ### First Check
> 
> - [X] I added a very descriptive title to this issue.
> - [X] I used the GitHub search to find a similar issue and didn't find it.
> - [X] I searched the FastAPI documentation, with the integrated search.
> - [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
> - [X] I already read and followed all the tutorial in the docs and didn't find an answer.
> - [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
> - [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
> - [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
> 
> ### Commit to Help
> 
> - [X] I commit to help with one of those options 👆
> 
> ### Example Code
> 
> ```python
> app = FastAPI()
> app.add_middleware(
>     TrustedHostMiddleware,
>     allowed_hosts=["localhost:8000"]
> )
> ```
> 
> 
> ### Description
> 
> Using the provided example code and making a request from localhost:8000 I get an "Invalid host header" response, even though looking at dev tools in chrome the header is `Host: localhost:8000`
> 
> If I change to `allowed_hosts=["localhost"]` it works fine.
> 
> I understand it's a trusted host, and not a trusted port, but seems strange to me that it says invalid host header even though thats exactly what the browser is sending in the host header.
> 
> Not sure if this is intended behaviour or not.
> 
> ### Operating System
> 
> Linux
> 
> ### Operating System Details
> 
> _No response_
> 
> ### FastAPI Version
> 
> 0.88.0
> 
> ### Python Version
> 
> 3.10.8
> 
> ### Additional Context
> 
> _No response_

Hey, man!

A hostname is the name of a server, which resolve to one (or multiple) IP addresses. You should not include the port as it have nothing do with hostname, at all.

**Comment by aybruhm** [comment]
Hello @blipk 

A hostname is the name of a server, which resolve to one (or multiple) IP addresses. You should not include the port as it have nothing do with hostname, at all.

**Comment by blipk** [comment]
@iudeen @aybruhm 
I understand that, but the Host header is passed with a port from the browser, and the TrustedHostMiddleware works on the basis of that header.

**Comment by aybruhm** [comment]
> @iudeen @aybruhm 
> I understand that, but the Host header is passed with a port from the browser, and the TrustedHostMiddleware works on the basis of that header.

The port should be specified when you want to run the server, not on the domain name(s). 

`uvicorn.run(
        "backend.apps.core.main:server", host="0.0.0.0", port=3000, reload=True
    )`

As you can see [_here_](https://fastapi.tiangolo.com/advanced/middleware/#trustedhostmiddleware), the domain names (without their port numbers) are been specified. The port that you pass to the uvicorn is where you'll be able to access the domain names. 

So if you set the server to run on port 3000. As long as you set the proxy port to use the above port, you should be able to access the domain names.

**Comment by blipk** [comment]
@aybruhm 
I have and can read the docs fine, doesn't make my question invalid just because you don't understand.

**Comment by jgould22** [comment]
The relevant bit of code is here https://github.com/encode/starlette/blob/master/starlette/middleware/trustedhost.py#L38

I agree since the Host header can include port (https://www.rfc-editor.org/rfc/rfc9110.html#field.host)  it should not be trimmed off before the check.

**Comment by blipk** [comment]
Thanks @jgould22 did not realise this was using starlette underneath.

Have posted the issue on their repo here: https://github.com/encode/starlette/issues/1997
